### DISCUSSION TITLE
How Type JSON?

### DISCUSSION THREAD

**Discussion by cawa-93** [prisma/prisma#13481] [Q&A]
I have a `Template` model. There are two columns
```prisma
model Template {
  id               Int         @id @default(autoincrement())
  layoutKind       LayoutKind
  layoutProperties Json
}

enum LayoutKind {
  Text
  Image
}
```
`layoutProperties` are strictly defined for each `layoutKind`. For Example if `layoutKind` is `Text` then `layoutProperties` should has type `{link: string}` and if `layoutKind` is `Image` then `layoutProperties` should has type `{src: string, alt: string}`.

How can I implement such a type system?

**Comment by nurul3101** [comment]
Hey @cawa-93 👋 

You could have two different nullable columns like `textLayoutProperties` and `imageLayoutProperties` and depending upon the `layoutKind` you insert in either of the columns.

```prisma
model Template {
  id                    Int        @id @default(autoincrement())
  layoutKind            LayoutKind
  textLayoutProperties  Json?
  imageLayoutProperties Json?
}

enum LayoutKind {
  Text
  Image
}
```

**Reply by cawa-93** [reply]
Unfortunately, this will not work for me, as there are more than two types of layouts. I provide only 2 just like example

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
